lib/middleware/user-data/user-data.js

```diff
@@ -1,365 +1,365 @@
-      delete data.COMPOSITE
-      data = removeNullEntries(data)
-
-      let uploadedFiles = []
-
-      const findUploadedFiles = (data) => {
-        if (Array.isArray(data) || typeof data !== 'object') {
-          return
-        }
-        Object.keys(data).forEach(key => {
-          const arr = data[key]
-          if (Array.isArray(arr)) {
-            const actualFiles = arr.filter(item => {
-              return item.fingerprint && item.url
-            }).map(item => ({
-              url: item.url,
-              mimetype: item.mimetype,
-              filename: item.originalname,
-              type: 'filestore'
-            }))
-            uploadedFiles = uploadedFiles.concat(actualFiles)
-          } else if (typeof arr === 'object') {
-            findUploadedFiles(arr)
-          }
-        })
-      }
-
-      findUploadedFiles(data)
-
-      return uploadedFiles
-    }
-  }
-  return Object.assign({}, additional, bundle)
-}
-
-external.loadUserData = async (req, res) => {
-  const userId = req.getUserId()
-  const userToken = req.getUserToken()
-  let data = {}
-
-  if (!req.newSession) {
-    try {
-      data = await loadData({ userId, userToken }, req.logger, req.session)
-    } catch (e) {
-      if (e.code === 404) {
-        error('User data not found', e)
-        return clearSessionAndRedirect(res)
-      }
-
-      error('Load user data error', e)
-      throw e
-    }
-  }
-
-  Object.defineProperty(req, 'user', {
-    writeable: false,
-    value: external.getUserDataMethods(data,
-      req.params || {},
-      {
-        saveData,
-        req,
-        res,
-        getUserId: req.getUserId,
-        getUserToken: req.getUserToken
-      })
-  })
-
-  Object.freeze(req.user)
-
-  if (req.newSession) {
-    const now = new Date()
-    const createdAt = now.toISOString()
-    const dataRetentionDuration = getInstanceProperty('service', 'dataRetentionDuration')
-    const expiresAt = moment(now).add(dataRetentionDuration, 'd').format('D MMMM YYYY')
-    req.user.setUserDataProperty('created', createdAt, 'meta')
-    req.user.setUserDataProperty('expires', expiresAt, 'meta')
-    await req.user.saveData()
-  }
-}
-
-module.exports = external
+require('@ministryofjustice/module-alias/register-module')(module)
+
+const moment = require('moment')
+
+const debug = require('debug')
+const error = debug('runner:user-data')
+
+debug.enable('runner:*')
+
+const {
+  getBracketNotationPath,
+  setBracketNotationPath,
+  unsetBracketNotationPath
+} = require('~/fb-runner-node/bracket-notation-path/bracket-notation-path')
+
+const { evaluate } = require('~/fb-runner-node/evaluate-condition/evaluate-condition')
+
+const cloneDeep = require('lodash.clonedeep')
+
+const { getInstanceByPropertyValue, getInstanceProperty } = require('~/fb-runner-node/service-data/service-data')
+
+const userDataStoreClient = require('~/fb-runner-node/client/user/datastore/datastore')
+
+const {
+  clearSessionCookie,

```

Comment:
I renamed this from `clearSession`. It's partner is `createSessionCookie` so it seemed odd that it didn't include the word "cookie" when, in fact, it doesn't clear the session -- it clears the session cookie